app/lib/theme/schemaFilteringHelpers.ts

```diff
@@ -22,14 +23,17 @@ export const getSchemaAndDataIncludingCalculatedValues = (
 
   for (const key of Object.keys(filteredSchema.properties)) {
     // null new data with undefined old data occurs when an optional form (e.g. budgets) is created but not filled
+
     if (
       formDataIncludingCalculatedValues?.[key] === null &&
       oldFormDataIncludingCalculatedValues?.[key] === undefined
     ) {
       delete filteredSchema.properties[key];
     } else if (
-      formDataIncludingCalculatedValues?.[key] ===
-      oldFormDataIncludingCalculatedValues?.[key]
+      isEqual(
+        formDataIncludingCalculatedValues?.[key],
+        oldFormDataIncludingCalculatedValues?.[key]
+      )
```

Comment:
No suggestions here, I'd like to know why we need to use the lodash here. Is it because we want to compare array objects?

Reply:
Yes, the anticipated funding per fiscal year and additional funding are arrays. If we just use `===`, they're never equal and we get a diff even when nothing has changed